fetch-in-react/my-app/src/Users.tsx

```diff
@@ -0,0 +1,73 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import { UserDetails } from './UserDetails';
+import { UserCard } from './UserCard';
+import './Users.css';
+
+export type User = {
+  id: number;
+  name: string;
+  email: string;
+  company: {
+    name: string;
+  };
+};
+export function Users() {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+  const [users, setUsers] = useState<User[]>([]);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    async function getUsers() {
+      try {
+        const url = 'https://jsonplaceholder.typicode.com/users';
+        const response = await fetch(url);
+        if (!response.ok) {
+          throw new Error(`Response status: ${response.status}`);
+        }
+        const json = await response.json();
```

Comment:
You should use type assertion here to make sure you have good type protection:

`const json = await response.json() as User[];`